### ISSUE TITLE
Build warnings for XML documentation comments

### ISSUE THREAD

**Issue by bradtglass** [SteveDunn/Vogen#278]
### Describe the bug

Generating Value Objects in a project that has the `GenerateDocumentationFile` option set to `true` in the project file creates multiple warnings during build:

```
CustomerId.g.cs(146,27): Warning CS1584 : XML comment has syntactically incorrect cref attribute '{item.UnderlyingTypeFullName}.ToString()'
CustomerId.g.cs(146,27): Warning CS1658 : Identifier expected. See also error CS1001.
CustomerId.g.cs(146,28): Warning CS1658 : Type parameter declaration must be an identifier not a type. See also error CS0081.
```

Looking into the warning a bit more - looks like a template isn't being expanded properly in the source generator for the `ToString()` method. The following is from the generated file for the example at the start of the README (`CustomerId`):
```c#
/// <inheritdoc cref="{item.UnderlyingTypeFullName}.ToString()" />
public readonly override global::System.String ToString() => Value.ToString();
```

### Steps to reproduce

1. Create a new project from the class library template
2. Install Vogen (version 3.0.3)
3. Replace `Class1` with the `CustomerId` example from the README:
``` c#
[ValueObject]
public partial struct CustomerId {
    // optional
    private static Validation Validate(int value) => value > 0 
        ? Validation.Ok 
        : Validation.Invalid("Customer IDs must be a positive number.");
}
```
4. Add `<GenerateDocumentationFile>true</GenerateDocumentationFile>` to the csproj `PropertyGroup`
5. Run `dotnet build`

### Expected behaviour

Source generator should not introduce warnings

**Comment by bradtglass** [comment]
I'm happy to take a look into fixing this too - doesn't seem like it will be a big thing to get working.

**Comment by SteveDunn** [comment]
Hi - thanks for bug report and the PR!  Much appreciated. I'll get that merged ASAP!

**Comment by SteveDunn** [comment]
@bradtglass - that has now been published in 3.0.4 on NuGet. Thanks again!
